### PR TITLE
More documentation on robottelo decorators. Fixes #3713

### LINKED ISSUE #3713: Add more documentation on decorators
Currently missing:

- [ ] run_in_one_thread
- [ ] tier{1,2,3,4}
- [ ] cacheable
- [ ] skip_if_not_set
- [ ] run_only_on

### PR DESCRIPTION
**Pull request by pondrejk**

### CODE REVIEW

File: docs/features/decorators.rst [hunk 1]
```diff
@@ -87,3 +87,86 @@ Arbitrary number versions can be passed as parameters. On second test both RHEL
...
+    from robottelo.decorators import cacheable
+
+    @cacheable
+    def make_role(options=None):
+        """create a role using ``hammer role create``"""
+
+run_only_on
+-----------
+
+``run_only_on`` skips test based on a server mode. It takes one parameter that is either 'sat' (to define Satellite specific tests) or 'sam' (to define tests specific to Subscription Asset Manager). For example::
```
Comment (rochacbruno) [inline docs/features/decorators.rst]: Looks like this is deprecated @omaciel can you confirm?
I found no reference to `run_only_on('sam')` in the code, are running builds against sam thing?

**Comment by rochacbruno** [comment]
ACK

Great to see more documentation!

File: docs/features/decorators.rst (hunk 1 above)
Comment (sghai) [inline docs/features/decorators.rst]: Yeah.. we were using this to run some tests on `SAM` product. But not sure if we are still testing that.
@sthirugn : could you please confirm ?

Comment (omaciel) [inline docs/features/decorators.rst]: If you had asked me this 2 weeks ago I'd say that we can get rid of this decorator... but *SAM* could be making a comeback, so let's leave it documented. @rochacbruno you can think of *SAM* as a 'light version' of *Satellite 6*.

**Comment by sghai** [comment]
ACK

File: docs/features/decorators.rst (hunk 1 above)
Comment (sthirugn) [inline docs/features/decorators.rst]: Yes what @omaciel said.  Although we are not doing any SAM specific testing now, we might do it in future.

**Pull request merged**